app/assets/javascripts/admin/product_import/controllers/import_form_controller.js.coffee

```diff
@@ -51,19 +52,26 @@ angular.module("admin.productImport").controller "ImportFormCtrl", ($scope, $htt
   $scope.start = () ->
     $scope.started = true
     total = ams_data.item_count
-    size = 50
-    $scope.chunks = Math.ceil(total / size)
-
-    i = 0
-
-    while i < $scope.chunks
-      start = (i*size)+1
-      end = (i+1)*size
-      if $scope.step == 'import'
-        $scope.processImport(start, end)
-      if $scope.step == 'save'
-        $scope.processSave(start, end)
-      i++
+    $scope.chunks = Math.ceil(total / $scope.batchSize)
+
+    # Process only the first batch.
+    $scope.processBatch($scope.step, 0, $scope.chunks)
+
+  $scope.processBatch = (step, batchIndex, batchCount) ->
+    start = (batchIndex * $scope.batchSize) + 1
+    end = (batchIndex + 1) * $scope.batchSize
+    withNextBatch = batchIndex + 1 < batchCount
+
+    promise = if step == 'import'
+      $scope.processImport(start, end)
+    else if step == 'save'
+      $scope.processSave(start, end)
+
+    processNextBatch = ->
+      $scope.processBatch(step, batchIndex + 1, batchCount)
+
+    # Process next batch whether or not processing of the current batch succeeds.
+    promise.then(processNextBatch, processNextBatch) if withNextBatch
```

Comment:
I felt like changing to isLastBatch and return early, please ignore if you prefer your version.
```suggestion
     $scope.processBatch = (step, batchIndex, batchCount) ->
    startRecord = (batchIndex * $scope.batchSize) + 1
    endRecord = (batchIndex + 1) * $scope.batchSize
    isLastBatch = batchCount < batchIndex + 1

    promise = if step == 'import'
      $scope.processImport(startRecord, endRecord)
    else if step == 'save'
      $scope.processSave(startRecord, endRecord)

    return if isLastBatch
    
    # Process next batch whether or not processing of the current batch succeeds.
    processNextBatch = ->
      $scope.processBatch(step, batchIndex + 1, batchCount)
    promise.then(processNextBatch, processNextBatch)
```

Reply:
I made the changes using `isLastBatch` instead and then returning early: https://github.com/openfoodfoundation/openfoodnetwork/pull/3387/commits/b6d8611359ed2cd701ee7997f095487c4bdcba92

Thought to note that I used `==` in the expression:

```
isLastBatch = batchCount == batchIndex + 1
```

Thanks, @luisramos0! :slightly_smiling_face: